packages/lit-html/src/directive-helpers.ts

```diff
@@ -121,7 +121,12 @@ export const insertPart = (
     );
   } else {
     const endNode = wrap(part._$endNode!).nextSibling;
-    if (endNode !== refNode) {
+    const parentChanged = part._$parent !== containerPart;
+    if (parentChanged) {
+      part._$reparentDisconnectables?.(containerPart);
```

Comment:
Any reason this is not the more straightforward `_$setParent` that sets the part's parent too?

Reply:
`_$reparentDisconnectables` only exists if there is a disconnectable somewhere in the part, as it's a patched-on API for pay-for-play code economy. As such, we can't rely on that being there to update `_$parent`, so the line below needs to do that unconditionally.